core/src/main/java/be/appwise/core/extensions/activity/ActivityExtensions.kt

```diff
@@ -3,17 +3,36 @@ package be.appwise.core.extensions.activity
 import android.Manifest
 import android.app.Activity
 import android.content.Context
+import android.content.pm.PackageManager
 import android.net.ConnectivityManager
 import android.util.DisplayMetrics
+import android.view.View
 import android.widget.EditText
+import androidx.annotation.ColorRes
 import androidx.annotation.RequiresPermission
+import androidx.core.content.ContextCompat
 import androidx.core.view.WindowInsetsCompat
 import androidx.core.view.WindowInsetsControllerCompat
-import be.appwise.core.extensions.snackBar
+import be.appwise.core.extensions.makeSnackbar
 
-@Deprecated("This is not the desired way to show/style a Snackbar, theming is easier to do globally in the themes.xml")
-fun Activity.snackBar(message: String, textColor: Int = android.R.color.white, viewID: Int = android.R.id.content) =
-    snackBar(findViewById(viewID), message, textColor).show()
+@Deprecated("This is not the desired way to show/style a Snackbar, theming is easier to do globally in the themes.xml (by changing 'surfaceColor'). Use 'showSnackbar(message, view, textColor)' for better results")
+fun Activity.snackbar(message: String, textColor: Int = android.R.color.white, viewID: Int = android.R.id.content) =
+    showSnackbar(message, findViewById(viewID), textColor)
+
+fun Activity.showSnackbar(message: String, view: View = findViewById(android.R.id.content), @ColorRes textColor: Int? = null) =
```

Comment:
I am confused. In the previous deprecated function it is mentioned that theming shouldn't be done this way but in the new function it is still done the same.

Reply:
Correct, theming should be done in `xml` to get better results.
However, I left the textColor option just for now to have a bit more flexibility.

The `showSnackbar` is just a shorter way of showing a snackbar than the default one.